internal/provisioner/plain/controllers/bundle_controller.go

```diff
@@ -206,27 +208,24 @@ func (r *BundleReconciler) ensureUnpackPod(ctx context.Context, bundle *rukpakv1
 		pod.Spec.AutomountServiceAccountToken = &automountServiceAccountToken
 		pod.Spec.Volumes = []corev1.Volume{
 			{Name: "util", VolumeSource: corev1.VolumeSource{EmptyDir: &corev1.EmptyDirVolumeSource{}}},
+			{Name: "manifests", VolumeSource: corev1.VolumeSource{EmptyDir: &corev1.EmptyDirVolumeSource{}}},
 		}
 		pod.Spec.RestartPolicy = corev1.RestartPolicyNever
-		if len(pod.Spec.InitContainers) != 1 {
-			pod.Spec.InitContainers = make([]corev1.Container, 1)
-		}
-		pod.Spec.InitContainers[0].Name = "install-unpacker"
-		pod.Spec.InitContainers[0].Image = r.UnpackImage
-		pod.Spec.InitContainers[0].ImagePullPolicy = corev1.PullIfNotPresent
-		pod.Spec.InitContainers[0].Command = []string{"cp", "-Rv", "/unpack", "/util/unpack"}
-		pod.Spec.InitContainers[0].VolumeMounts = []corev1.VolumeMount{{Name: "util", MountPath: "/util"}}
-
-		if len(pod.Spec.Containers) != 1 {
-			pod.Spec.Containers = make([]corev1.Container, 1)
-		}
-		if bundle.Spec.Source.Image != nil {
-			pod.Spec.Containers[0].Name = bundleUnpackContainerName
-			pod.Spec.Containers[0].Image = bundle.Spec.Source.Image.Ref
-			pod.Spec.Containers[0].Command = []string{"/bin/unpack", "--bundle-dir", "/"}
-			pod.Spec.Containers[0].VolumeMounts = []corev1.VolumeMount{{Name: "util", MountPath: "/bin"}}
+
+		switch bundle.Spec.Source.Type {
+		case rukpakv1alpha1.SourceTypeImage:
+			pod = bundleImagePod(pod, *bundle.Spec.Source.Image, r.UnpackImage)
```

Comment:
It's possible that the `bundle.Spec.Source.Image` is not set.  It may be better that `bundleImagePod` validates the source and return error if it's not set.

Reply:
I think this is something I'd expect to be enforced in CRD schema validation (which doesn't exist right now but we're targeting that work in a follow-up), but maybe we should double dip into validation in code too?

Reply:
The `bundleGitRepoPod` does validation but `bundleImagePod` doesn't.  I see a little inconsistency.

Reply:
That's true, I would expect this validation at the schema level but we can add some validation here as well. I think it should be a follow-up since it's not directly related to this PR.